M2/Macaulay2/packages/Msolve.m2

```diff
@@ -0,0 +1,527 @@
+msolvePresent := run ("type msolve >/dev/null 2>&1") === 0
+-- msolveVersion := if msolvePresent then ... -- no way to get version info!!!
+msolveVersionNeeded := "0.6.4"
+msolvePresentAndModern := msolvePresent -- and match("^[0-9.]+$",msolveVersion) and msolveVersion >= msolveVersionNeeded
+
+newPackage(
+	"Msolve",
+	Version => "1.24.05", 
+    	Date => "July 2024",
+    	Authors => {{Name => "Martin Helmer", 
+		  Email => "mhelmer@ncsu.edu", 
+		  HomePage => "http://martin-helmer.com/"}, {Name => "Mike Stillman", 
+		  Email => "mike@math.cornell.edu", 
+		  HomePage => "https://math.cornell.edu/michael-e-stillman"},{Name => "Anton Leykin", 
+		  Email => "leykin@math.gatech.edu", 
+		  HomePage => "https://antonleykin.math.gatech.edu/"}},
+    	Headline => "An interface to the msolve package (https://msolve.lip6.fr/) which computes Groebner Basis and does real root isolation",
+    	AuxiliaryFiles => true,
+	DebuggingMode => false,       
+	OptionalComponentsPresent => msolvePresent
```

Comment:
I generally don't appreciate packages that run an external program upon being loaded (which also ends up happening when `Macaulay2Doc` is being installed). This is perhaps beyond the scope of this PR, but I wish `OptionalComponentsPresent` could be set _after_ `newPackage`, perhaps even in the call to `beginDocumentation`, so that loading could stop there if the optional component is not present.